### PR TITLE
Hw3_t02

### PR DESCRIPTION
**Pull request by mirmozavr**

### CODE REVIEW

File: hw3/hw3_t02.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    data = hashlib.md5(str(value).encode()).digest()  # noqa: S303
+    return sum(struct.unpack("<" + "B" * len(data), data))
+
+
+def slow_calculate_batch(*nums: int) -> None:
+    for num in nums:
+        slow_calculate(num)
+
+
+def slow_calc_500_numbers() -> None:
```
Comment (Cjkjvfnby) [inline hw3/hw3_t02.py]: I don't know if this code produces the expected result, need a test for that.

File: hw3/hw3_t02.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    if __name__ == "__main__":
```
Comment (Cjkjvfnby) [inline hw3/hw3_t02.py]: Please check an example in the documentation about using these lines.

File: hw3/hw3_t02.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        procs = []
+        for tens in range(50):
+            batch = tuple((tens * 10) + digit for digit in range(10))
+            proc = Process(target=slow_calculate_batch, args=batch)
```
Comment (Cjkjvfnby) [inline hw3/hw3_t02.py]: You are not limited by low-level primitives in this task.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw3/hw3_t02.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def slow_calculate(value: int) -> int:
+    time.sleep(random.randint(1, 3))  # noqa: S311
+    data = hashlib.md5(str(value).encode()).digest()  # noqa: S303
+    return sum(struct.unpack("<" + "B" * len(data), data))
+
+
+def slow_calc_500_numbers() -> list:
+    with Pool(processes=50) as pool:
```
Comment (Cjkjvfnby) [inline hw3/hw3_t02.py:23]: Why have you chosen 50?

File: hw3/tests/test_hw3_t02.py [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+    2750,
+    2763,
+    2766,
+    2785,
+    2810,
+]
+
+
+def test_slow_calc_500_numbers():
+    assert sorted(slow_calc_500_numbers()) == pre_calculated_data_sorted
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t02.py]: What was the task requirement?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw3/tests/test_hw3_t02.py (hunk 5 above)
Comment (mirmozavr) [reply hw3/tests/test_hw3_t02.py]: To calculate the total sum

File: hw3/hw3_t02.py (hunk 4 above)
Comment (mirmozavr) [reply hw3/hw3_t02.py:23]: 50 works well under 30 seconds, 10 is too slow, 100 doesn't finish at all.

File: hw3/tests/test_hw3_t02.py (hunk 5 above)
Comment (Cjkjvfnby) [reply hw3/tests/test_hw3_t02.py]: Sometimes my question is the only question, sometimes it is a call to action. 

Does your function calculate the sum?

File: hw3/hw3_t02.py (hunk 4 above)
Comment (Cjkjvfnby) [reply hw3/hw3_t02.py:23]: > 100 doesn't finish at all

What do you mean?   They are freezes without any message?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: hw3/hw3_t02.py (hunk 4 above)
Comment (mirmozavr) [reply hw3/hw3_t02.py:23]: No messages, no errors, just endless running.

File: hw3/tests/test_hw3_t02.py (hunk 5 above)
Comment (mirmozavr) [reply hw3/tests/test_hw3_t02.py]: Yeah, I corrected my code. I'll push that changes after figuring out a number of processes.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
And the final question: how many numbers should we sum?

**Comment by mirmozavr** [comment]
Must be 501? Was that inclusive 0 to 500?

**Comment by Cjkjvfnby** [comment]
`all numbers starting from 0 to 500` sounds like 0 and 500 should be present.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**